game-core/src/main/java/games/strategy/triplea/ui/screen/Tile.java

```diff
@@ -15,80 +15,66 @@
-   * internal mutex to prevent tiles being drawn mid-refresh.
-   */
-  public Object getMutex() {
-    return mutex;
+    return isDirty && !isDrawing.get();
   }
 
   /** Returns the image representing this tile, re-rendering it first if the tile is dirty. */
-  public Image getImage(final GameData data, final MapData mapData) {
-    if (isDirty) {
-      synchronized (mutex) {
-        isDrawing = true;
-        final Graphics2D g = (Graphics2D) image.getGraphics();
-        g.setRenderingHint(RenderingHints.KEY_RENDERING, RenderingHints.VALUE_RENDER_QUALITY);
-        g.setRenderingHint(
-            RenderingHints.KEY_ALPHA_INTERPOLATION,
-            RenderingHints.VALUE_ALPHA_INTERPOLATION_QUALITY);
-        g.setRenderingHint(
-            RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BICUBIC);
-        g.setRenderingHint(
-            RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
-        draw(g, data, mapData);
-        g.dispose();
-        isDrawing = false;
-      }
+  public void drawImage(final GameData data, final MapData mapData) {
+    if (isDirty && !isDrawing.getAndSet(true)) {
+      final Image backImage = Util.newImage(TILE_SIZE, TILE_SIZE, true);
+      final Graphics2D g = (Graphics2D) backImage.getGraphics();
+      g.setRenderingHint(RenderingHints.KEY_RENDERING, RenderingHints.VALUE_RENDER_QUALITY);
+      g.setRenderingHint(
+          RenderingHints.KEY_ALPHA_INTERPOLATION, RenderingHints.VALUE_ALPHA_INTERPOLATION_QUALITY);
+      g.setRenderingHint(
+          RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BICUBIC);
+      g.setRenderingHint(
+          RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
+      draw(g, data, mapData);
+      g.dispose();
+      image = backImage;
+      isDrawing.set(false);
     }
-    return image;
   }
 
-  /**
-   * This image may not reflect our current drawables. Use getImage() to get a correct image
-   *
-   * @return the image we currently have.
-   */
-  public Image getRawImage() {
-    synchronized (mutex) {
-      return image;
-    }
+  /** This image may not reflect our current drawables. */
+  public Image getImage() {
+    return image;
   }
 
   private void draw(final Graphics2D g, final GameData data, final MapData mapData) {
     final AffineTransform original = g.getTransform();
     // clear
     g.setColor(Color.BLACK);
-    g.fill(new Rectangle(0, 0, TileManager.TILE_SIZE, TileManager.TILE_SIZE));
-    final Queue<IDrawable> queue = new PriorityQueue<>(contents);
-    while (!queue.isEmpty()) {
-      queue.remove().draw(bounds, data, g, mapData);
-      // Make sure we don't mess up other draws
-      g.setTransform(original);
+    g.fill(new Rectangle(0, 0, TILE_SIZE, TILE_SIZE));
+    synchronized (mutex) {
+      final Queue<IDrawable> queue = new PriorityQueue<>(contents);
```

Comment:
As you can see the only thing that "mutex" does now is synchronize the access to `contents` and `isDirty` as a single operation.
Not sure if it would work without any unintended problems if we made contents a concurrent collection of some sort and removed the mutex. Currently calling `new PriorityQueue<>(contents)` is a constant time operation, changing the type of contents would make it `O(n * log n)`.
We could however use a concurrent set where the draw level is compared, and if even we compare the hashCodes or something in order not to "remove" "duplicate" tiles with the same draw level as others.